roles/aodh/tasks/main.yml

```diff
@@ -69,17 +81,22 @@
 - name: start aodh services
   service: name={{ item }} state=started
   with_items:
-    - aodh-api
-    - aodh-evaluator
-    - aodh-listener
-    - aodh-notifier
+    - "{{ openstack_meta.aodh.services.aodh_api[os].name }}"
+    - "{{ openstack_meta.aodh.services.aodh_evaluator[os].name }}"
+    - "{{ openstack_meta.aodh.services.aodh_listener[os].name }}"
+    - "{{ openstack_meta.aodh.services.aodh_notifier[os].name }}"
 
 - name: add aodh-expirer to cron.daily
   file: src=/usr/local/bin/aodh-expirer dest=/etc/cron.daily/aodh-expirer state=link owner=root group=root mode=0755
+  when: os == 'ubuntu'
```

Comment:
you can use ternary to abstract the path difference and justhave one common task rather than 2 task by OS